### ISSUE TITLE
Multi-Grained Scanning

### ISSUE THREAD

**Issue by Colin1592** [LAMDA-NJU/Deep-Forest#78]
Hi! I am learning deep forest, and I come across a question. That is when we do sliding, say we have 400-dim raw input features, and then we generate 301 instances with size of 100, for the first instance, we train a forest, and then the second instance comes, so the second instance is input into the first forest or used to train another forest? In other words, in multi-grained scanning part, we train 1 forest or 301 forests?

**Comment by xuyxu** [comment]
Hi @Colin1592, I am not sure that if I understand your question correctly :-(

Suppose that you are given a training data of size `(N, 400)` (`N` the number of training instances), then after sliding, the training data turns into `(N*301, 100)`. This new training data will be used to train multiple forest models, as detailed in the original paper.

**Comment by xuyxu** [comment]
Closed due to inactivity.

**Comment by Colin1592** [comment]
If the input data is (N, 400), after sliding, the training data turns into (N*301, 100), if we use the new training data to train a forest, we just get a 3-dim classification distribution.  But In my opinion, I think the after sliding, training data will be 301*(N, 100), and we can get 301 * 3-dim classification. But if the case is what i think, different fetures will train differnt forest. So I am confused.

**Comment by MAbderrahmane** [comment]
Hello @Colin1592 I think the author mean 301 new sampled training set instead of 301 instances
Suppose X : Training set ( 1000 instances, 400 features), Window size = 100
After Sliding window procedure we will get a 301 * (1000 instances, 100 features)
